spec/models/miq_widget_spec.rb

```diff
@@ -539,8 +536,7 @@ def add_dashboard_for_user(db_name, userid, group)
         @widget.visibility[:roles] = "_ALL_"
         MiqWidgetSet.destroy_all
         user = FactoryBot.create(:user, :userid => 'alone', :miq_groups => [@group2])
-        ws = FactoryBot.create(:miq_widget_set, :set_data_with_one_widget, :name => "default", :userid => user.userid, :read_only => true)
-        ws.add_member(@widget)
+        ws = FactoryBot.create(:miq_widget_set, :name => "default", :userid => user.userid, :read_only => true, :widget_id => @widget.id)
```

Comment:
again, I can't comment on the right line, but `queue_generate_content` calls `available_for_group` (`group_hash = group_hash_visibility_agnostic.select { |k, _v| available_for_group?(k) }`) which calls `has_visibility?(:roles, group.miq_user_role_name) || has_visibility?(:groups, group.description)` which is passing nil as a second arg into `def has_visibility?(key, value)` which breaks on `visibility[key].include?(value)`